### PR TITLE
Add wind direction widget with logic

### LINKED ISSUE #118: Wind direction UI - JS
Week 10

Redoing #62 using JS

### PR DESCRIPTION
**Pull request by robertyac**
Screenshot 2024-03-10 at 11 42 54 PM

Should be good to go. Test it out!

closes #118

### CODE REVIEW

**Comment by Davis245** [comment]
Do I need to do anything special to get this to work? Right now I am unable to load the compass.

Screenshot 2024-03-11 at 12 03 08 PM

Also, I really like the design you settled on! However, I think the double arrow is a little bit misleading on which way it is pointing. I would suggest keeping the grey arrow head and removing the red arrow head because wind direction is based off of where it is coming from.

**Comment by robertyac** [comment]
Should work? Refresh cache?

**Comment by Davis245** [comment]
I've tried that. I think I'll just wait until someone else tries it to see if they have similar issues or if I'm doing something wrong.

**Comment by robertyac** [comment]
Did You select a day before march 5th? there's no recent data that's been uploaded

Edit: you selected "today" try selecting a day before march 5th

**Comment by robertyac** [comment]
I can remove the grey arrow head easily lmk
Screenshot 2024-03-11 at 6 27 07 PM

**Comment by Davis245** [comment]
Haha oops, selected an older date and it works!

I like it better with just the one arrow head. Also, something to think about, wind direction is kind of counter intuitive in that the direction given is the direction the wind is coming from. This is because when you measure wind direction you stand facing the wind and the direction you're facing is the one recorded. For example, in the last screenshot where the wind direction is 358 degrees the wind is originating in the north and blowing south. Therefore, I think the arrow should point south in this situation.

If we end you using the direction of the arrow described above I think it might be confusing to the user because they may ask the question, "that arrow is clearly not pointing to 358 degrees". I think a solution to this problem could be to add a few more arrows to the compass to make it look more like wind.

IMG_94B2FE49EC08-1

I do not know how difficult this is to implement, so if it is too much of a hassle it is not needed.

**Comment by robertyac** [comment]
I see, i interpreted it wrong then. I assumed that’s the direction the wind
was going towards not coming from. I can just reverse the arrow in that
case then.

I don’t want to spend too much longer on this though I rather implement
some other stuff like the testing and such. If you guys want me to put more
arrows it may just take away from other issues let me know if we think it’s
worth it.

Either way I’ll just reverse the arrow direction.

**Comment by robertyac** [comment]
We should develop paper prototypes for the other widgets so we're all on board with their visuals since I did spend a decent amount of time already with this one.

Just so we are all on board before spend so much time on all the widgets and we are satisfied with their designs.

**Comment by Davis245** [comment]
Fair enough. The testing is definitely more important.

**Comment by robertyac** [comment]
Screenshot 2024-03-11 at 10 05 49 PM

This a little better? It makes a little more sense now

**Review by Davis245 — APPROVED** [review]
Perfect

File: Site/bc_weather_station_dashboard/website/static/map.js [hunk 1]
```diff
@@ -106,7 +106,13 @@ function updateDataHTML(currentStationData) {
...
     if (currentStationData.HOURLY_WIND_DIRECTION) {
-        document.getElementById('wind-direction').innerHTML = currentStationData.HOURLY_WIND_DIRECTION + "&deg;";
+        // Get the wind direction in degrees
+        var windDirectionDegrees = currentStationData.HOURLY_WIND_DIRECTION;
+        // Update the text display append to the wind direction widget
+        document.getElementById('wind-direction').innerHTML = "<h5>Wind Direction " + windDirectionDegrees + "&deg;</h5>";
+        // Create a new WindArrow with the updated wind direction
+        var windArrow = new WindArrow(windDirectionDegrees);
+        windArrow.draw();
     }
```
Comment (cmcahoon01) [inline Site/bc_weather_station_dashboard/website/static/map.js:116]: Do we want to move these to their own functions?

**Review by cmcahoon01 — APPROVED** [review]

File: Site/bc_weather_station_dashboard/website/static/map.js (hunk 1 above)
Comment (robertyac) [reply Site/bc_weather_station_dashboard/website/static/map.js:116]: Maybe in the future? right now map.js updates all the station elements linking them to a specific station and time such as temp, wind direction etc...

**Pull request merged**